src/gui/plotWidget.cpp

```diff
@@ -262,3 +262,38 @@ void Plot::clearSelection() {
         graph->setSelection(QCPDataSelection{});
     }
 }
+
+void Plot::convertToNewUnit(UnitValue current, UnitValue next) {
+    // Check for a valid unit
+    if(current != UnitValue::KN && current != UnitValue::KGF && current != UnitValue::LBF) {
+        return;
+    }
```

Comment:
This shouldn't need to be checked. A better check would be if the unit to be converted too is already the current unit.

Reply:
Check isn't needed anymore.
The check wether the unit has changed is already done before the function call.
https://github.com/linescaleGUI/linescaleGUI/blob/a05e133a64655be318c9eb4186151b2e73f7c3f2/src/gui/plotWidget.h#L84-L90